dts: arm: st: h7: add ITCM and DTCM memory regions for STM32H7

iabdalkader opened the pull request:
Add ITCM and DTCM memory region definitions for the STM32H7.

ITCM: 64 KB @ 0x00000000
DTCM: 128 KB @ 0x20000000

JarmouniA commented on dts/arm/st/h7/stm32h747Xi_m7.dtsi:
```diff
@@ -13,6 +13,18 @@
 	};
 
+	itcm: memory@0 {
+		compatible = "zephyr,memory-region", "arm,itcm";
```
nit: compatibles go from, left to right, most to least specific, `arm,i/dtcm` is more specific than `zephyr,memory-region` (but one could say they are not comparable to begin with, hence the "nit").

erwango reviewed (commented):
From the search I've just made, all STM32H7 based soc use 64/128K ITCM/DTCM.
Unless I'm wrong, it would be better to put it directly in h7.dtsi (and delete node from m4 cores).

iabdalkader committed:
dts: arm: st: h7: add ITCM and DTCM memory regions for STM32H7

Add ITCM and DTCM memory region definitions to the common STM32H7
devicetree.

ITCM: 64 KB @ 0x00000000
DTCM: 128 KB @ 0x20000000

Signed-off-by: Ibrahim Abdalkader <i.abdalkader@gmail.com>

iabdalkader commented:
> it would be better to put it directly in h7.dtsi (and delete node from m4 cores).

We'll also need to delete any itcm/dtcm defined by _m7 files. This will be a much bigger change, 10+ files, is that okay?

iabdalkader committed:
dts: arm: st: h7: delete ITCM/DTCM nodes from M4 devicetree

Delete the ITCM and DTCM nodes from all M4-specific devicetree.

Signed-off-by: Ibrahim Abdalkader <i.abdalkader@gmail.com>

erwango reviewed (changes requested):
Thanks!

But please merge all your commits (at the minimum #1 and #3) as having them split will break bisectability (it won't compile due to duplicated nodes)

erwango reviewed (approved):
I could not find a H7 that is not compatible with this setting from datasheet.

mathieuchopstm reviewed (approved):
Seems correct per [ST Product Selector](https://www.st.com/en/microcontrollers-microprocessors/stm32h7-series/products.html) (which seems to have a bug making no TCM appear despite some products having it... glanced at a few datasheets too)
